src/Api/Validator.php

```diff
@@ -189,6 +189,13 @@ private function check_boolean(Shape $shape, $value)
 
     private function check_string(Shape $shape, $value)
     {
+        if ($shape['jsonvalue']
+            && !is_resource($value)
+            && (is_object($value) || is_array($value))
```

Comment:
Validation should be isolated from serialization. If I run my input data through the validator, then it should ensure that the values I provided are valid based on the model. This is a separate process that runs before serialization. You can also get much better error messages from validation than from a serialization error because the validator tracks error context whereas serialization does not.